cinema/views.py

```diff
@@ -101,16 +104,36 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None) -> Response:
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
You should pass `raise_exception=True` to is_valid, it returns `Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)` automatically.
```suggestion
        serializer.is_valid(raise_exception=True)
        serializer.save()
        return Response(serializer.data)
```